src/codegen/compile.ml

```diff
@@ -3170,51 +3534,84 @@ module MakeCompact (Num : BigNumType) : BigNumType = struct
     try_unbox I32Type (fun _ -> match n with
         | 32 | 64 -> G.i Drop ^^ Bool.lit true
         | 8 | 16 ->
-          compile_bitand_const Int32.(logor 1l (shift_left minus_one (n + 1))) ^^
+          (* Please review carefully! *)
```

Comment:
can remove

Reply:
I change the comment to explain the shift (and removed the logor 1l)